### PR TITLE
fix(dev-infra): call release build function directly

### PR DESCRIPTION
**Pull request by josephperrott**
Previously the release build function was called indirectly through `ng-dev release build`,
now it is called as a function directly.  This change prevents an issue in which
the compiled ng-dev release tool calls the uncompiled version or vice versa.

Additionally, generates a local copy of the build-worker file to allow it to be loaded
at runtime in a forked process.

### CODE REVIEW

**Comment by devversion** [comment]
IIRC we decided to go through the "ng-dev release" command so that it also works for older version branches which could have different release packages or different build process.

With the new changes (that still seem to fulfil that requirement), we rely on a "build-worker.js" file to exist in every version branch. I think having a set of commands (that are guaranteed to be implemented in a version branch; as we discussed in the past) is more flexible, future-proof and easier to understand. Expecting every version branch to have a "build-worker.js" file seems like something that could break more easily. We also have another "external command" for setting the NPM dist tags right?

File: dev-infra/release/build/build.spec.ts [hunk 1]
```diff
@@ -6,72 +6,60 @@
...
+       const buildResultPromise = buildReleaseOutput();
+       mockChildProcess.emit('message', npmPackages);
+       mockChildProcess.emit('exit', 0);
+       await expectAsync(buildResultPromise).toBeResolvedTo(npmPackages);
+     });
 
-  it('should print built packages as JSON if `--json` is specified', async () => {
-    const writeSpy = spyOn(process.stdout, 'write');
-    await invokeBuild({json: true});
+  it('should invoke configured build packages function and resolve to null if not packages are built',
```
Comment (gkalpak) [inline dev-infra/release/build/build.spec.ts:35]:
```suggestion
  it('should invoke configured build packages function and resolve to null if no packages are built',
```

**Review by gkalpak — CHANGES_REQUESTED** [review]
Some minor comments and questions. Otherwise lgtm :+1:

**Comment by josephperrott** [comment]
I had forgotten about everything Paul mentioned in his comment, closing this in favor of #40012.

**Pull request closed without merging**